src/main/java/com/aws/greengrass/mqtt/bridge/clients/IoTCoreClient.java

```diff
@@ -143,89 +121,260 @@ public void publish(com.aws.greengrass.mqtt.bridge.model.MqttMessage message) {
                 .payload(message.getPayload())
                 .qos(QualityOfService.AT_LEAST_ONCE)
                 .build();
-        iotMqttClient.publish(publishRequest);
+        // pubacks are not propagated
+        iotMqttClient.publish(publishRequest).exceptionally(e -> {
+            LOGGER.atWarn().cause(e).kv(LOG_KEY_TOPIC, message.getTopic())
+                    .log("Failed to publish message");
+            return null;
+        });
     }
 
     @Override
     public void updateSubscriptions(Set<String> topics,
                                     @NonNull Consumer<com.aws.greengrass.mqtt.bridge.model.MqttMessage>
                                             messageHandler) {
-        synchronized (subscribeLock) {
-            if (subscribeFuture != null) {
-                subscribeFuture.cancel(true);
-            }
-
-            this.messageHandler = messageHandler;
-            this.toSubscribeIotCoreTopics = new HashSet<>(topics);
-            LOGGER.atDebug().kv("topics", topics).log("Subscribing to IoT Core topics");
-
-            Set<String> topicsToRemove = new HashSet<>(subscribedIotCoreTopics);
-            topicsToRemove.removeAll(topics);
-            topicsToRemove.forEach(s -> {
-                try {
-                    unsubscribeFromIotCore(s);
-                    LOGGER.atDebug().kv(TOPIC, s).log("Unsubscribed from topic");
-                    subscribedIotCoreTopics.remove(s);
-                } catch (InterruptedException | ExecutionException | TimeoutException e) {
-                    LOGGER.atError().kv(TOPIC, s).setCause(e).log("Unable to unsubscribe");
-                    // If we are unable to unsubscribe, leave the topic in the set
-                    // so that we can try to remove next time.
-                }
-            });
-
-            Set<String> topicsToSubscribe = new HashSet<>(topics);
-            topicsToSubscribe.removeAll(subscribedIotCoreTopics);
-
-            subscribeFuture = executorService.submit(() -> subscribeToTopicsWithRetry(topicsToSubscribe));
-        }
+        this.messageHandler = messageHandler;
```

Comment:
do we need to change the message handler all the time? That seems like it should only be set once.